src/test/ui/rfc1598-generic-associated-types/shadowing.rs

```diff
@@ -0,0 +1,44 @@
+// Copyright 2012 The Rust Project Developers. See the COPYRIGHT
+// file at the top-level directory of this distribution and at
+// http://rust-lang.org/COPYRIGHT.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+#![feature(generic_associated_types)]
+
+//FIXME(#44265): The lifetime shadowing and type parameter shadowing
```

Comment:
These errors around shadowing seems like expected behavior to me. We disallow shadowing of lifetimes in part because it tends to confuse everybody =)

Reply:
That is what I intended and I clarified the text a bit. There *should* be errors.